web/controllers/user_controller.ex

```diff
@@ -72,6 +72,15 @@ defmodule Streamr.UserController do
     render(conn, "show.json-api", data: Plug.current_resource(conn))
   end
 
+  def update(conn, %{"user" => user_params}) do
+    changeset = User.changeset(conn.assigns.current_user, user_params)
+
+    case Repo.update(changeset) do
+      {:ok, user} -> render(conn, "show.json-api", data: user)
+      {:error, errors} -> conn |> put_status(422) |> render("errors.json-api", data: errors)
```

Comment:
Not that I know anything about elixir, but I like this one-liner style for case statements.